backend/src/main/java/kernel360/techpick/core/auth/service/CustomOAuth2Service.java

```diff
@@ -0,0 +1,74 @@
+package kernel360.techpick.core.auth.service;
+
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+import kernel360.techpick.core.auth.model.OAuth2UserInfo;
+import kernel360.techpick.core.config.OAuth2AttributeConfigProvider;
+import kernel360.techpick.core.feature.user.User;
+import kernel360.techpick.feature.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class CustomOAuth2Service extends DefaultOAuth2UserService {
+
+	private final UserRepository userRepository;
+	private final OAuth2AttributeConfigProvider configProvider;
+
+	@Override
+	public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+		String provider = userRequest.getClientRegistration().getRegistrationId();
+		var oAuth2User = super.loadUser(userRequest);
+		Map<String, Object> attributes = getAttributes(oAuth2User, provider);
+		var oAuth2UserInfo = new OAuth2UserInfo(provider, attributes);
+
+		if (!userRepository.existsBySocialProviderId(oAuth2UserInfo.getName())) {
+			saveOAuth2UserInfo(oAuth2UserInfo);
+		}
+		return oAuth2UserInfo;
+	}
+
+	private Map<String, Object> getAttributes(OAuth2User oAuth2User, String provider) {
+		Map<String, String> config = configProvider.getAttributeConfig(provider);
```

Comment:
p3: Map 을 감싸는 Class를 하나 만들어서 쓰면 어떨지?

```suggestion
		AttributeConfig config = configProvider.getAttributeConfig(provider);
```

Reply:
`@ConfigurationProperties`으로 `oauth2-config.yaml`에서 값을 가져와야하므로 편의를 위해 `Map`으로 설정